### PR TITLE
Add the role-has-responsible-party constraint

### PR DESCRIPTION
**Pull request by DimitriZhurkin**
# Committer Notes

Add the `role-has-responsible-party` constraint.

### All Submissions:

- [ ] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]
Can you adjust the `-FAIL.yaml` and `INVALID.xml` test to include a a role not bound to a specific party? I do not see that in this pull request.

Otherwise, it is looking good.

**Comment by DimitriZhurkin** [comment]
> Can you adjust the `-FAIL.yaml` and `INVALID.xml` test to include a a role not bound to a specific party? I do not see that in this pull request.
> 
> Otherwise, it is looking good.


AJ,

There are already two roles without responsible parties in the `ssp-all-INVALID.xml` (see below).

I did not want to add another role on the premise that it might complicate matters for the team in the long run.
```
   <role id="system-admin">
      <title>System Administrator</title>
    </role>
    <role id="asset-owner">
      <title>Asset Owner</title>
    </role>
```

**Comment by aj-stein-gsa** [comment]
> There are already two roles without responsible parties in the `ssp-all-INVALID.xml` (see below).
> 
> I did not want to add another role on the premise that it might complicate matters for the team in the long run.

I understand your point, but I would like to see that one of the two roles in the valid version properly bound to a `responsible-party/@role-id` and one that is not, only to make it incredibly clear by comparing valid and invalid what is missing, also perhaps a comment where it could be.

**Comment by DimitriZhurkin** [comment]
> > There are already two roles without responsible parties in the `ssp-all-INVALID.xml` (see below).
> > I did not want to add another role on the premise that it might complicate matters for the team in the long run.
> 
> I understand your point, but I would like to see that one of the two roles in the valid version properly bound to a `responsible-party/@role-id` and one that is not, only to make it incredibly clear by comparing valid and invalid what is missing, also perhaps a comment where it could be.
> 
> **UPDATE:** An example is in this [GitHub gist for more tangible explanation.](https://gist.github.com/aj-stein-gsa/15e2c2dac7bbc64e001052499742dfcc/revisions).

In `ssp-all-INVALID.xml`, added a role without a responsible party. Also, revised the `role-has-responsible-party-FAIL.yaml` file.

The commit created a new pull request.

File: src/validations/constraints/unit-tests/role-has-responsible-party-FAIL.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Driver for the invalid role-has-responsible-party constraint unit test.
+test-case:
+  name: The invalid role-has-responsible-party constraint unit test.
+  description: Test that SSP metadata roles do not have responsible parties associated with them.
+  content: ../content/ssp-all-INVALID.xml
```
Comment (Gabeblis) [inline src/validations/constraints/unit-tests/role-has-responsible-party-FAIL.yaml:5]: `ssp-all-INVALID.xml` has been removed (https://github.com/GSA/fedramp-automation/pull/713). This constraint will need it's own `ssp-role-has-responsible-party-INVALID.xml` file, and the yaml fail test file will need to reference the new `-INVALID.xml` file. You can manually make these changes, or delete the existing pass/fail yaml files for this constraint and run `npm run constraint role-has-responsible-party` to automatically scaffold the `-INVALID.xml`, with the minimum necessary context to trigger the constraint, and the corresponding pass/fail yaml files.

**Pull request closed without merging**